examples/browser-api-playground/src/components/Layout/PageLayout.tsx

```diff
@@ -1,18 +1,36 @@
-import { PropsWithChildren } from "react";
+import { PropsWithChildren, useEffect, useState } from "react";
 import Header from "./Header";
-import { EmbedConfigContext } from "../../EmbedConfigContext";
+import { EmbedConfigContext, sdkOptionsKey } from "../../EmbedConfigContext";
 import useConfigStore from "../../useConfigStore";
 import { Layout } from "antd";
 import { Content } from "antd/es/layout/layout";
 
 const PageLayout = ({ children }: PropsWithChildren) => {
+  const [ready, setReady] = useState(false)
   const contextValue = useConfigStore()
 
+  useEffect(() => {
+    try {
+      if (!contextValue.config?.[sdkOptionsKey]?.source) {
+        throw new Error('No source URL provided')
+      } else {
+        const script = document.createElement('script')
+        script.src = contextValue.config[sdkOptionsKey].source
+        script.onload = () => {
+          setReady(true)
+        }
+        document.head.appendChild(script)
+      }
+    } catch (e) {
+      alert('Please provide a source URL in the "SDK Options" section of the config')
+    }
+  }, [])
```

Comment:
Currently, the config is loaded synchronously in `useConfigStore`. But if its made async then this useEffect would throw an alert because at initial render the config would be undefined. 
This can be fixed in followup though.

Reply:
interesting, I imagine we will have a default config no matter sync or async, so it should be rare that the source is undefined ? 

but ya, once we move to async, the logic here and probably other places will require some changes

Reply:
In the `useConfigStore`, we merge the `defaultConfig` with the config stored in the session storage. This currently happens synchronously.

Reply:
ya I mean even if we move to async, the default config is likely still holds and set synchronously - but ya let's update when we do move to async